tools/integration-tests/tester/tests/mana/mana_test.go

```diff
@@ -45,3 +48,62 @@ func TestManaPersistence(t *testing.T) {
 	require.Greater(t, manaAfter.Access, 0.0)
 	require.Greater(t, manaAfter.Consensus, 0.0)
 }
+
+func TestAPI(t *testing.T) {
+	numPeers := 2
+	n, err := f.CreateNetwork("mana_TestAPI", 0, 0)
+	require.NoError(t, err)
+	defer tests.ShutdownNetwork(t, n)
+
+	// create peers
+	peers := make([]*framework.Peer, numPeers)
+	for i := 0; i < numPeers; i++ {
+		peer, err := n.CreatePeer(framework.GoShimmerConfig{
+			Mana:               true,
+			SyncBeaconFollower: true,
+		})
+		require.NoError(t, err)
+		peers[i] = peer
+	}
+
+	allowedPeer := peers[0]
+	allowedID := base58.Encode(allowedPeer.Identity.ID().Bytes())
+	disallowedPeer := peers[1]
+	disallowedID := base58.Encode(disallowedPeer.Identity.ID().Bytes())
+
+	// faucet
+	faucet, err := n.CreatePeer(framework.GoShimmerConfig{
+		Faucet:                            true,
+		Mana:                              true,
+		ManaAllowedAccessFilterEnabled:    true,
+		ManaAllowedConsensusFilterEnabled: true,
+		ManaAllowedAccessPledge:           []string{allowedID},
+		ManaAllowedConsensusPledge:        []string{disallowedID},
+		SyncBeacon:                        true,
+	})
+
+	require.NoError(t, err)
+	err = n.WaitForAutopeering(2)
+	require.NoError(t, err)
+
+	time.Sleep(10 * time.Second)
+
+	addrBalance := make(map[string]map[balance.Color]int64)
+	faucetAddrStr := faucet.Seed.Address(1).String()
+	addrBalance[faucetAddrStr] = make(map[balance.Color]int64)
+	addrBalance[allowedPeer.Address(0).String()] = make(map[balance.Color]int64)
+	addrBalance[disallowedPeer.Address(0).String()] = make(map[balance.Color]int64)
+
+	// get faucet balances
+	unspentOutputs, err := faucet.GetUnspentOutputs([]string{faucetAddrStr})
+	require.NoErrorf(t, err, "could not get unspent outputs on %s", faucet.String())
+	addrBalance[faucetAddrStr][balance.ColorIOTA] = unspentOutputs.UnspentOutputs[0].OutputIDs[0].Balances[0].Value
+
+	// pledge mana to allowed pledge
+	fail, _ := tests.SendIotaTransaction(t, faucet, allowedPeer, addrBalance, 100, 1, 0, allowedPeer.Identity.ID(), allowedPeer.Identity.ID())
+	require.False(t, fail)
+
+	// pledge mana to disallowed pledge
+	fail, _ = tests.SendIotaTransaction(t, faucet, disallowedPeer, addrBalance, 100, 1, 0, disallowedPeer.Identity.ID(), disallowedPeer.Identity.ID())
```

Comment:
why are we not spending from the second address of the faucet? Didn't the previous transaction above move the funds there?

Reply:
Yes. We should